File: domain/src/main/java/hanghae/domain/entity/Seat.java
package hanghae.domain.entity;

import jakarta.persistence.*;
import lombok.NoArgsConstructor;

@Entity
@Table(
        name = "seats",
        uniqueConstraints = @UniqueConstraint(columnNames = {"screen_id", "seat_name"})
)
@NoArgsConstructor
public class Seat {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long seatId;

    private String seatName;

    @ManyToOne
    @JoinColumn(name = "screen_id")
    private Screen screen;
}

File: cinema/src/main/kotlin/cinema/domain/Seat.kt
package cinema.domain

import com.redis.cinema.domain.Cinema
import com.redis.utils.BaseEntity
import jakarta.persistence.*

@Table(name = "seat")
@Entity
data class Seat(

    @Column(name = "row")
    val row: String,

    @Column(name = "column")
    val column: String,

    @ManyToOne
    @JoinColumn(name = "cinema_id")
    val cinema: Cinema,

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "seat_id")
    val id: Long? = null
): BaseEntity() {

}